test/plug/conn_test.exs

```diff
@@ -405,6 +405,20 @@ defmodule Plug.ConnTest do
     assert get_resp_header(conn, "x-body") == ["CHUNK"]
   end
 
+  test "chunk/2 runs before_chunk callbacks" do
```

Comment:
Let's call `register_before_chunk/2` *twice* in this test, send slightly different messages (such as `{:before_chunk, 1, chunk}` and then with `2`), and make sure that the order the callbacks are executed in is correct.

Reply:
Thanks for catching this